finetuner-workflow/finetuner/finetuner.py

```diff
@@ -972,6 +979,10 @@ def evaluate(
         )
     )
 
+# Calculate how many warmup steps we need to do using the warmup ratio.
+training_steps = int((args.epochs * len(train_dataset)) / (args.gradients * bs * world_size))
+warmup_steps = int(args.warmup_ratio * training_steps)
+
```

Comment:
This calculation can be skipped by specifying `warmup_ratio` directly in the `TrainingArguments` constructor (see other comment)
```suggestion
```

Reply:
Oh wow didn't know that